### DISCUSSION TITLE
How does genhtml detect common filename prefix

### DISCUSSION THREAD

**Discussion by my-umd** [linux-test-project/lcov#299] [General]
(Using lcov 2.1)
I noticed that the automatically detected common filename prefix is not accurate when there are two forward slashes (at the beginning of the pathname).  E.g., `SF://export/home/tomcat/TeamCity/Agent/src/internal/filename`.  When I ran `genhtml`, I see this from my console:
`Found common filename prefix "/"`
While the actual common filename pattern is `//export/home/tomcat/TeamCity/Agent/src/internal`.
I understand that the extra forward slash is not correct here.  But seems `genhtml` will not correctly identify the common pattern when it is present.  I did a test by removing the extra forward slash from my trace file, and verified that `genhtml` works correctly in this case.  Not sure if this is a bug.

**Comment by henry2cox** [comment]
A feature - but not a bug.
lcov/genhtml/geninfo do not try to be clever with respect to files and paths.  If string is _identical_, the file is the same.  If string differs - then not.
I suspect that it found `/` as the common prefix because some paths in your input data started with `//` and some with `/` - and genhtml gave up on commoning when it saw the first difference.

For this specific issue, you can use `genhtml --substitute 's#//#/#' ...` to tell the tool to munge the path before processing (you may need the quotes so your shell doesn't think it saw a comment character).

**Comment by my-umd** [comment]
Ok, thanks @henry2cox.
